apps/backend/src/evaluations/dto/create-evaluation.dto.ts

```diff
@@ -1,26 +1,17 @@
 import {ICreateEvaluation} from '@heimdall/interfaces';
-import {
-  IsArray,
-  IsBoolean,
-  IsNotEmpty,
-  IsObject,
-  IsOptional,
-  IsString
-} from 'class-validator';
+import {IsArray, IsIn, IsNotEmpty, IsOptional, IsString} from 'class-validator';
 import {CreateEvaluationTagDto} from '../../evaluation-tags/dto/create-evaluation-tag.dto';
 
 export class CreateEvaluationDto implements ICreateEvaluation {
   @IsNotEmpty()
   @IsString()
   readonly filename!: string;
 
-  @IsNotEmpty()
-  @IsObject()
   readonly data!: Record<string, any>;
 
   @IsNotEmpty()
-  @IsBoolean()
-  readonly public!: boolean;
+  @IsIn([true, false, 'true', 'false'])
+  public!: boolean | string;
```

Comment:
Can the DTO constructor handle this conversion instead of allowing strings here>